src/init.cpp

```diff
@@ -1029,8 +1079,12 @@ bool AppInit2(boost::thread_group& threadGroup)
                 pcoinscatcher = new CCoinsViewErrorCatcher(pcoinsdbview);
                 pcoinsTip = new CCoinsViewCache(pcoinscatcher);
 
-                if (fReindex)
+                if (fReindex) {
                     pblocktree->WriteReindexing(true);
+                    //If we're reindexing in prune mode, wipe away all our block and undo data files
```

Comment:
This doesn't seem to be what "reindexing" means, so I think this may be against user expectations. Perhaps introduce a new option ("-reset" ?), which wipes everything. Then you can just say that reindexing is not compatible with pruned mode.